examples/multiple_windows/lib/app/tooltip_button.dart

```diff
@@ -0,0 +1,176 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: invalid_use_of_internal_member
+// ignore_for_file: implementation_imports
+
+import 'package:flutter/material.dart';
+import 'package:flutter/src/widgets/_window.dart';
+import 'models.dart';
+
+/// Tracks global rect of an [Element].
+class ElementPositionTracker {
+  ElementPositionTracker(this.element) {
+    _register(this);
+  }
+
+  final BuildContext element;
+
+  /// Returns current global rect for tracked element or `null` if not available.
+  Rect? getGlobalRect() {
+    final rect = _getGlobalRect();
+    _lastReportedRect = rect;
+    return rect;
+  }
+
+  /// Disposes this tracker causing it to stop providing updates. The tracker is
+  /// automatically disposed when tracked element is unmounted.
+  void dispose() {
+    _unregister(this);
+  }
+
+  /// Callback invoked every time the global position of the tracked element changes
+  /// compared to last result of [getGlobalRect].
+  void Function(Rect rect)? onGlobalRectChange;
+
+  static bool _persistentCallbackRegistered = false;
+  static final _trackers = <ElementPositionTracker>[];
+
+  static void _register(ElementPositionTracker tracker) {
+    if (!_persistentCallbackRegistered) {
+      _persistentCallbackRegistered = true;
+      WidgetsBinding.instance.addPersistentFrameCallback((_) {
+        _update();
```

Comment:
Why is it necessary to update all trackers and not just this one? I want to make sure that it's necessary to use this singleton pattern with static state. It sometimes makes things hard to test, etc.